src/durable-cache/src/lib.rs

```diff
@@ -293,6 +300,43 @@ impl<C: DurableCacheCodec> DurableCache<C> {
             }
         }
     }
+
+    async fn compare_and_append<I>(
+        &mut self,
+        updates: I,
+        expected_upper: u64,
+        new_upper: u64,
+    ) -> Result<(), UpperMismatch<u64>>
+    where
+        I: IntoIterator<Item = ((C::KeyCodec, C::ValCodec), u64, i64)>,
+    {
+        self.write
+            .compare_and_append(
+                updates,
+                Antichain::from_elem(expected_upper),
+                Antichain::from_elem(new_upper),
+            )
+            .await
+            .expect("usage should be valid")?;
+
+        // Lag the shard's upper by 1 to keep it readable.
+        let downgrade_to = Antichain::from_elem(new_upper.saturating_sub(1));
+
+        // The since handle gives us the ability to fence out other downgraders using an opaque token.
+        // (See the method documentation for details.)
```

Comment:
Are we using the `SinceHandle` just to fence out other downgraders, or to hold back compaction, or both?

Reply:
Neither, actually! We're using it to advance compaction :)

Reply:
To drive compaction forward. My understanding is that we don't provide the same guarantees around compaction without a critical since handle.